lessons/python/inttopics/inttopics.ipynb

```diff
@@ -0,0 +1,672 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## List Comprehensions\n",
+    "\n",
+    "Let's say we want to get the original scores, out of 40. Is there a built-in method we could use in order to do the same thing to every item in a list?"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false,
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "dir(mylist)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Apparently not. Perhaps a for loop is the way to go? Let's test that out."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false,
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "for i in range(len(scores)):\n",
+    "    scores[i] = scores[i] * 0.4\n",
+    "    \n",
+    "print(scores)     "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Whoa, that's not right!"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "print(scores) "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Looks like Python imported the data as a list of strings, save for that one 89% we appended.\n",
+    "\n",
+    "We need to find a way to change those strings to integers instead, and then multiply them by 0.4."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "# this is one way?\n",
+    "for num in scores:\n",
+    "    if type(num) is str:\n",
+    "        num = int(num)\n",
+    "        num = num * 0.4\n",
+    "    else:\n",
+    "        pass"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "# or - we could use a list comprehension!\n",
+    "\n",
+    "scores = [int(num)*0.4 for num in scores] \n",
+    "\n",
+    "# conditionals can also be included in list comprehensions - this is valid syntax\n",
+    "\n",
+    "scores = [int(num)*0.4 for num in scores if type(num) is str] "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "For large operations, list comprehensions are also faster than using `list.append`. When the latter is a part of a loop, Python looks up the list in memory at every single iteration, which can really slow things down for large lists. List comprehensions, on the other hand, do not require the look up operation. They're also far more visually compact while conveying the same amount of information."
```

Comment:
Nifty! I'll admit it's a little confusing at first glance, but it can't hurt to put it in there for the sake of completeness. Thanks for the tip!